File: app/src/main/java/ch/ralena/quizapp/objects/Quiz.java
package ch.ralena.quizapp.objects;

import ch.ralena.quizapp.objects.Question;

public class Quiz {
	private Question mQuestion;
	private int mTotalTries;
	private int mCorrectTries;
	public final static int MAX_TRIES = 20;

	public Quiz() {
		mQuestion = new Question();
		mTotalTries = 0;
		mCorrectTries = 0;
	}

	public boolean nextQuestion() {
		if (mTotalTries == MAX_TRIES)
			return false;
		else {
			mQuestion.changeEquation();
			return true;
		}
	}

	public boolean checkAnswer(int answer) {
		boolean correct = answer == mQuestion.getCorrect();
		mTotalTries++;
		if (correct)
			mCorrectTries++;
		return correct;
	}

	// getters

	public String getQuestion() {
		return mQuestion.getQuestion();
	}

	public int getTotalTries() {
		return mTotalTries;
	}

	public int getCorrectTries() {
		return mCorrectTries;
	}

	public int getAnswer(int i) {
		return mQuestion.getAnswer(i);
	}
}

File: app/src/main/java/ch/ralena/quizapp/objects/Quiz.kt
package ch.ralena.quizapp.objects

class Quiz {
	var question: Question = Question()
	var totalTries: Int = 0
	var correctTries: Int = 0

	fun nextQuestion(): Boolean {
		question = Question()
		return totalTries < Constants.MAX_TRIES
	}

	fun checkAnswer(index: Int): Boolean {
		val correct = index == question.correctAnswerIndex
		totalTries++
		if (correct) correctTries++
		return correct
	}
}